packages/app-types/test/default.test.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import firebase from '@firebase/app';
+
+/**
+ * Assert `firebase.apps` is an array of Firebase apps
+ *
+ * i.e. We should be able to access the public members
+ * of the App from inside of an individual app.
+ */
+firebase.apps.forEach(app => {
+  const _name: string = app.name;
+  const _options: Object = app.options;
+  const _delete: Promise<any> = app.delete();
+});
```

Comment:
I'm not exactly sure what is going on with this "test" file...  Since we're not actually asserting anything, I *think* this is just compile-time verification (do we even execute this file? or just compile it?)...  So it's basically a way for us to write against the .d.ts and sort of sanity-check that the type definitions are really what we think they are?  

I find the value of this somewhat questionable (it doesn't provide much extra value than just reviewing the .d.ts directly), but if that is the intent, perhaps we should at least add a comment to the top of the file explaining the intent and perhaps change the "Assert that" wording, since there are no assertions here.

Note that if the intention is to verify that the runtime behavior matches our type definitions, I don't think it's very successful since e.g. there won't be any apps and so this forEach callback won't be run, and even if it was, if there was a mistake in the FirebaseApp typings (e.g. there isn't actually a .name property) this code likely wouldn't detect it since it's not asserting anything...  (similarly the firebase.SDK_VERSION regex isn't asserted to pass, so it wouldn't test anything, etc.)

Reply:
As I mentioned in the PR description, this is the "testing" that is done in the DefinitelyTyped repo (the only other repo I'm aware of that actively does curated types). I'll pull the wording, but the purpose of this file (and the others that need to be fleshed out) is to illustrate expected valid use cases of the API and make sure that they are considered type-safe.